r21 | chrism | 2005-09-07 15:34:00 +0200 (Wed, 07 Sep 2005) | 3 lines

fix pk7_mime.c to prevent garbled messages because of to early memory
    free (closes: #310184)

Index: openssl-0.9.8k/crypto/pkcs7/pk7_mime.c
===================================================================
--- openssl-0.9.8k.orig/crypto/pkcs7/pk7_mime.c	2008-11-05 19:36:48.000000000 +0100
+++ openssl-0.9.8k/crypto/pkcs7/pk7_mime.c	2009-07-19 17:13:53.000000000 +0200
@@ -335,9 +335,9 @@
 
 		if(strcmp(hdr->value, "application/x-pkcs7-signature") &&
 			strcmp(hdr->value, "application/pkcs7-signature")) {
-			sk_MIME_HEADER_pop_free(headers, mime_hdr_free);
 			PKCS7err(PKCS7_F_SMIME_READ_PKCS7,PKCS7_R_SIG_INVALID_MIME_TYPE);
 			ERR_add_error_data(2, "type: ", hdr->value);
+			sk_MIME_HEADER_pop_free(headers, mime_hdr_free);
 			sk_BIO_pop_free(parts, BIO_vfree);
 			return NULL;
 		}
